liquibase-core/src/main/java/liquibase/util/SpringBootFatJar.java

```diff
@@ -0,0 +1,12 @@
+package liquibase.util;
+
+public class SpringBootFatJar {
+    public static String getPathForResource(String path) {
+        String[] components = path.split("!");
+        if (components.length == 3) {
+            return String.format("%s%s", components[1].substring(1), components[2]);
```

Comment:
Why is the components[1] part required?
This creates following issue for me at the moment. Let's say I have my changeSet file `main/resources/db/changes/foo.sql`. The changeLog looks like this:
```
<databaseChangeLog
...
    <includeAll path="db/changes" />
...
</databaseChangeLog>
```

If I run the application as a Spring Boot packaged fat-jar, then filename of the aforemention name is `BOOT-INF/classes/db/foo.sql`, but when I run Boot application using `gradle bootRun` or using IDE run-configuration, then the path for the same changeSet file becomes `db/changes/foo.sql`.

This inconsistency makes Liquibase treat that same file as different changeSets, depending on the path they are accessed by, and re-run the SQL script.

Reply:
When the `ClassLoaderResourceAccessor` iterates over the entries in the zip file and compares their start, it needs to include both component[1] and component[2] in order to find them in this check:

`                        if (entry.getName().startsWith(path)) {`

The entries in the zip look like `/WEB-INF/classes/mypackage/foo` but with just component[2] it will only check against `/foo` and not actually find anything. At least that's what I'm seeing.

My full resolved path is:
`jar:file:/Users/username/sandbox/projectName/project/build/libs/myproject-7.5.0.war!/WEB-INF/classes!/migrations/`

Reply:
You are right though, the name needs to be considered without the middle part from a changeSet name perspective so perhaps the right change is to filter that part out of the entry name when looking for a match in the Jar file.